docling/backend/msexcel_backend.py

```diff
@@ -0,0 +1,316 @@
+                table_bounds, visited_cells = self.find_table_bounds(
+                    sheet, ri, rj, visited
+                )
+
+                visited.update(visited_cells)  # Mark these cells as visited
+                tables.append(table_bounds)
+
+        return tables
+
+    def find_table_bounds(
+        self,
+        sheet: Worksheet,
+        start_row: int,
+        start_col: int,
+        visited: set[Tuple[int, int]],
+    ):
+        """
+        Determine the bounds of a compact rectangular table.
+        Returns:
+        - A dictionary with the bounds and data.
+        - A set of visited cell coordinates.
+        """
+        _log.info("find_table_bounds")
+
+        max_row = start_row
+        max_col = start_col
+
+        # Expand downward to find the table's bottom boundary
+        while (
+            max_row < sheet.max_row - 1
+            and sheet.cell(row=max_row + 2, column=start_col + 1).value is not None
+        ):
+            max_row += 1
+
+        # Expand rightward to find the table's right boundary
+        while (
+            max_col < sheet.max_column - 1
+            and sheet.cell(row=start_row + 1, column=max_col + 2).value is not None
+        ):
+            max_col += 1
+
+        # Collect the data within the bounds
+        data = []
+        visited_cells = set()
+        for ri in range(start_row, max_row + 1):
+            # row_data = []
+            for rj in range(start_col, max_col + 1):
+
+                cell = sheet.cell(row=ri + 1, column=rj + 1)  # 1-based indexing
+
+                # Check if the cell belongs to a merged range
+                row_span = 1
+                col_span = 1
+                for merged_range in sheet.merged_cells.ranges:
+                    if (ri + 1, rj + 1) in merged_range:
+                        # Calculate the spans
+                        row_span = merged_range.max_row - merged_range.min_row + 1
+                        col_span = merged_range.max_col - merged_range.min_col + 1
+                        break
+
+                data.append(
+                    {
+                        "row": ri - start_row,
+                        "col": rj - start_col,
+                        "cell": cell,
+                        "row_span": row_span,
+                        "col_span": col_span,
+                    }
+                )
+
+                # Mark all cells in the span as visited
+                for span_row in range(ri, ri + row_span):
+                    for span_col in range(rj, rj + col_span):
+                        visited_cells.add((span_row, span_col))
+
+        return {
+            "beg_row": start_row,
+            "beg_col": start_col,
+            "end_row": max_row,
+            "end_col": max_col,
+            "num_rows": max_row + 1 - start_row,
+            "num_cols": max_col + 1 - start_col,
+            "data": data,
+        }, visited_cells
```

Comment:
Let's use pydantic classes instead of raw dictionaries, as this keeps way easier to maintain.
Also I think several of these keys (e.g. "beg_row", "beg_col", ...) appear to be unused.

Reply:
done